src/SqlAsyncEnumerable.cs

```diff
@@ -26,8 +25,12 @@ internal class SqlAsyncEnumerable<T> : IAsyncEnumerable<T>
         /// </exception>
         public SqlAsyncEnumerable(SqlConnection connection, SqlAttribute attribute)
         {
-            this._connection = connection ?? throw new ArgumentNullException(nameof(connection));
+            this.Connection = connection ?? throw new ArgumentNullException(nameof(connection));
             this._attribute = attribute ?? throw new ArgumentNullException(nameof(attribute));
+            if (!string.IsNullOrEmpty(this.Connection.ConnectionString))
```

Comment:
Is there a reason you're adding this check here? If the connection string is empty then nothing will work - so having it fail the Open call seems fine rather than just silently skipping the open

Reply:
TestNullCurrentValueEnumerator calls the constructor without any connection string and it fails, so I added that check.

Reply:
I would suggest refactoring that test to be like TestNullArgumentsSqlAsyncEnumerableConstructor and assert that it throws. Without a connection we can't do anything so it seems reasonable to me to throw immediately in that case instead of just skipping the open and later on failing.